cmd/createtree/main.go

```diff
@@ -72,15 +76,21 @@ func createTree(ctx context.Context) (*trillian.Tree, error) {
 
 	conn, err := grpc.Dial(*adminServerAddr, grpc.WithInsecure())
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("failed to dial %v: %v", *adminServerAddr, err)
 	}
 	defer conn.Close()
 
-	tree, err := trillian.NewTrillianAdminClient(conn).CreateTree(ctx, req)
-	if err != nil {
-		return nil, err
+	for {
```

Comment:
Personally I would push the retries to the bash scripts. Up to you.

Reply:
Then there's no way to distinguish between error types -- `Unavailable` is the only one [documented as explicitly retriable](https://github.com/grpc/grpc-go/blob/master/codes/codes.go#L86).